WordPress/Classes/Extensions/AbstractPost+fixedMediaLocalURL.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+extension AbstractPost {
+    /// If a post is in the failed state it can contain references to local files.
+    /// When updating the app through the App Store the local paths can change.
+    /// This will fix any outdated local path with the correct one.
+    ///
+    func fixedMediaLocalURL() {
+        guard isFailed,
+            var content = self.content else {
+            return
+        }
+
+        media.forEach { media in
+            guard !media.hasRemote else {
+                return
+            }
+
+            extractFilenameAndAbsolutePath(from: media)?.forEach { filename, absolutePath in
+                content = content.replacingMatches(of: "src\\s*=\\s*\"(file://.[^\"]+?/\(NSRegularExpression.escapedPattern(for: filename)))\"", with: "src=\"\(absolutePath)\"")
+            }
+        }
+
+        self.content = content
+    }
+
+    private func extractFilenameAndAbsolutePath(from media: Media) -> [(String, String)]? {
```

Comment:
The method signature here could use a touch-up —  there's no way for this method to actually ever return `nil`, so can we get rid of the `Optional` here?

Additionally, I'd love to have the tuple parameters be named, like `-> [(localURL: String, absoluteLocalURL: String` or so.